### DISCUSSION TITLE
Socket proxy error- Error: no_glibc

### DISCUSSION THREAD

**Discussion by dreamworks** [nextcloud/all-in-one#4521] [Questions] [answered]
Hi,

I am little lost; i configured the nextcloud-aio using docker (the following command)

docker run --init --sig-proxy=false --name nextcloud-aio-mastercontainer --restart always --publish 8080:8080 --env APACHE_PORT=11000 --env APACHE_IP_BINDING=0.0.0.0 --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config --volume /var/run/docker.sock:/var/run/docker.sock:ro nextcloud/all-in-one:latest

Everything works as expected, but as soon as I wanted to open up an office document I get the error "socket connection could not be established..."

In the detailed log it says the following:

Didn't find urlsrc for mimetype application/vnd.openxmlformats-officedocument.wordprocessingml.document in this WOPI discovery response: Socket proxy error - Error: no_glibc

I use apache as reverse proxy (see config below). Any hints are very much appreciated.. Not sure why it doesn' run out of the box..

```
<VirtualHost *:443>
    ServerName nextcloud.xxx.de
    # Reverse proxy based on https://httpd.apache.org/docs/current/mod/mod_proxy_wstunnel.html
    RewriteEngine On
    ProxyPreserveHost On
    AllowEncodedSlashes NoDecode



    ProxyPass / http://localhost:11000/ nocanon
    ProxyPassReverse / http://localhost:11000/

    RewriteCond %{HTTP:Upgrade} websocket [NC]
    RewriteCond %{HTTP:Connection} upgrade [NC]
    RewriteCond %{THE_REQUEST} "^[a-zA-Z]+ /(.*) HTTP/\d+(\.\d+)?$"
    RewriteRule .? "ws://localhost:11000/%1" [P,L]

    # Enable h2, h2c and http1.1
    Protocols h2 h2c http/1.1

    # Solves slow upload speeds caused by http2
    H2WindowSize 5242880

    # TLS
    SSLEngine               on
    SSLProtocol             -all +TLSv1.2
    SSLCipherSuite          ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-A
    SSLHonorCipherOrder     off
    SSLSessionTickets       off
    SSLCertificateFile /root/.acme.sh/xxx.de/xxx.de.cer
    SSLCertificateKeyFile /root/.acme.sh/xxx.de/xxx.de.key
    SSLCertificateChainFile /root/.acme.sh/xxx.de/ca.cer


    # Disable HTTP TRACE method.
    TraceEnable off
    <Files ".ht*">
        Require all denied
    </Files>

    # Support big file uploads
    LimitRequestBody 0
</VirtualHost>

```

**Comment by szaimen** [comment]
Hi, can you follow https://github.com/nextcloud/all-in-one/discussions/1358?

**Reply by dreamworks** [reply]
Pl. see my answer/finding below.. hope this helps...

**Comment by dreamworks** [comment, marked as answer]
Hi,

I found the issue. It has nothing to do with nextcloud, but actually with apache. I used Centos v7 with apache v2.4.39 and docker to deploy nextcloud aio, BUT it seems that the Centos 7 apache is missing mod_proxy_wstunnel, and therefore the websocket calls are failing.

After I migrated my installation on Rocky Linux v9 (apache v2.4.57) it's working like a charm with the exact same reverse proxy configration mentioned .. 

So actually it's an apache issue, although the error message within nextcloud is definitly misleading, as it has nothing to do with nextcloud at all.

Hope my finding also helps some other out there.. !

**Reply by andrewhamili** [reply]
> Hi,
> 
> I found the issue. It has nothing to do with nextcloud, but actually with apache. I used Centos v7 with apache v2.4.39 and docker to deploy nextcloud aio, BUT it seems that the Centos 7 apache is missing mod_proxy_wstunnel, and therefore the websocket calls are failing.
> 
> After I migrated my installation on Rocky Linux v9 (apache v2.4.57) it's working like a charm with the exact same reverse proxy configration mentioned ..
> 
> So actually it's an apache issue, although the error message within nextcloud is definitly misleading, as it has nothing to do with nextcloud at all.
> 
> Hope my finding also helps some other out there.. !

Will enabling proxy_wstunnel won't help? I am currently using Apache/2.4.41 (Ubuntu) with proxy_wstunnel enabled
